src/Security/SecurityControllerBuilder.php

```diff
@@ -80,4 +87,45 @@ public function addLogoutMethod(ClassSourceManipulator $manipulator): void
         );
         $manipulator->addMethodBuilder($logoutMethodBuilder);
     }
+
+    public function addFormLoginMethod(ClassSourceManipulator $manipulator, string $loginTemplatePath): void
+    {
+        $methodBuilder = $manipulator->createMethodBuilder('login', Response::class, false);
+        $methodBuilder->addAttribute($manipulator->buildAttributeNode(Route::class, ['path' => '/login', 'name' => 'app_login']));
+
+        $this->addUseStatements($manipulator, [Route::class]);
+
+        $methodBuilder->addParam((new Param('authenticationUtils'))->setType('AuthenticationUtils'));
+
+        $contents = file_get_contents(\dirname(__DIR__).'/Resources/skeleton/security/formLogin/_LoginMethodBody.tpl.php');
+
+        $manipulator->addMethodBody($methodBuilder, $contents, ['template_path' => $loginTemplatePath]);
+        $manipulator->addMethodBuilder($methodBuilder);
+    }
+
+    public function addJsonLoginMethod(ClassSourceManipulator $manipulator, ClassNameDetails $userClass): void
+    {
+        $methodBuilder = $manipulator->createMethodBuilder('apiLogin', JsonResponse::class, false);
+
+        $methodBuilder->addAttribute($manipulator->buildAttributeNode(Route::class, ['path' => '/api/login', 'name' => 'app_api_login']));
+
+        $this->addUseStatements($manipulator, [Route::class, $userClass->getFullName(), CurrentUser::class]);
+
+        $methodBuilder->addParam(
+            (new Param('user'))
+                ->setType(new NullableType($userClass->getShortName()))
+                ->addAttribute(new Attribute(new Name('CurrentUser')))
+        );
```

Comment:
This probably needs to change to `$this->getUser()` for better "portability" across all versions of Symfony, DoctrineBundle, FWEXtra, etc?

Reply:
Should we bother making that conditional based on the Symfony Version? Or perhaps we do that later when say 6.4 or 7 come out for less of a headache for us